### PR TITLE
Adding oVirt Node image plugin and docs

### PR DESCRIPTION
**Pull request by lzap**
Big README split, see here:

https://github.com/lzap/foreman_discovery/tree/ovirt-node

### CODE REVIEW

File: README.ovirt_image.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    DEFAULT menu
+    PROMPT 0
+    MENU TITLE PXE Menu
+    TIMEOUT 200
+    TOTALTIMEOUT 6000
+    ONTIMEOUT discovery
+
+    LABEL discovery
+    MENU LABEL Foreman Discovery
+    KERNEL boot/oVirt-vmlinuz
```
Comment (domcleal) [inline README.ovirt_image.md]: Above you name them discovery-\* but refer here to oVirt-*

File: README.ovirt_image.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    APPEND rootflags=loop initrd=boot/oVirt-initrd.img root=live:/foreman.iso rootfstype=auto ro rd.live.image rd.live.check rd.lvm=0 rootflags=ro crashkernel=128M elevator=deadline max_loop=256 rd.luks=0 rd.md=0 rd.dm=0 foreman.ip=192.168.122.1:3000 selinux=0 stateless
+
+Note the long line and do not forget to press "Build PXE Default" button afterwards.
+Also SELinux must be turned off currently.
```
Comment (domcleal) [inline README.ovirt_image.md]: on the image.  I misread this as on Foreman as it's mentioned with the button above.

File: README.ovirt_image.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    PROMPT 0
+    MENU TITLE PXE Menu
+    TIMEOUT 200
+    TOTALTIMEOUT 6000
+    ONTIMEOUT discovery
+
+    LABEL discovery
+    MENU LABEL Foreman Discovery
+    KERNEL boot/oVirt-vmlinuz
+    APPEND rootflags=loop initrd=boot/oVirt-initrd.img root=live:/foreman.iso rootfstype=auto ro rd.live.image rd.live.check rd.lvm=0 rootflags=ro crashkernel=128M elevator=deadline max_loop=256 rd.luks=0 rd.md=0 rd.dm=0 foreman.ip=192.168.122.1:3000 selinux=0 stateless
```
Comment (domcleal) [inline README.ovirt_image.md]: Using port 80 for the example foreman.ip might be best for production use

File: README.tcl_image.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# Tine Core Linux discovery image
+
+This page describes how to download, build and configure discovery image.
+
+# Building the Discovery PXE Image
+
+There is a rake task for the discovery image. It requires Ruby 1.8, but should
+appropriately detect and use ruby1.8 even if the system default is 1.9. If you
+have trouble building it, try a ruby1.8-only system.
```
Comment (domcleal) [inline README.tcl_image.md]: Is that really true?  Worked for me on a ruby193 system..

File: README.tcl_image.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# Tine Core Linux discovery image
+
+This page describes how to download, build and configure discovery image.
```
Comment (domcleal) [inline README.tcl_image.md:3]: Download instructions would be great as they're currently missing.

File: README.ovirt_image.md (hunk 1 above)
Comment (lzap) [inline README.ovirt_image.md]: Oh yeah I was renaming that and missed this one.

File: README.ovirt_image.md (hunk 2 above)
Comment (lzap) [inline README.ovirt_image.md]: Cheap, yeah, fixing.

File: README.ovirt_image.md (hunk 3 above)
Comment (lzap) [inline README.ovirt_image.md]: This is copied from Greg's README. Yeah, changing here.

File: README.tcl_image.md (hunk 5 above)
Comment (lzap) [inline README.tcl_image.md:3]: Yeah, Greg any notes for the three commands above? I can add the link and change 3000 port in the TCL README.

**Comment by lzap** [comment]
All fixed, not sure about the Ruby 1.8 there.

File: README.tcl_image.md (hunk 4 above)
Comment (GregSutcliffe) [inline README.tcl_image.md]: It used to be true, I tried to improve it :)

File: README.tcl_image.md (hunk 5 above)
Comment (GregSutcliffe) [inline README.tcl_image.md:3]: I guess download only really applies to getting it from the directory on deb.tf.o? In which case they're on the line below.

File: README.tcl_image.md (hunk 4 above)
Comment (domcleal) [inline README.tcl_image.md]: I'd remove the disclaimer then

File: README.tcl_image.md (hunk 5 above)
Comment (domcleal) [inline README.tcl_image.md:3]: yup

**Comment by GregSutcliffe** [comment]
README looks good. I'll run through the instructions tomorrow and give it a test.

File: README.tcl_image.md (hunk 4 above)
Comment (GregSutcliffe) [inline README.tcl_image.md]: I wasn't 100% sure. thanks for the confirmation.

File: extra/ovirt-node-plugin-foreman/README.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    git clone https://github.com/theforeman/foreman_discovery.git
+    cd foreman_discovery/extra/ovirt-node-plugin-foreman
+    ./autogen.sh
+    make rpms
+
+Now, let's modify the base ISO image and install the plugin there.
+
+    cd $OVIRT_NODE_BASE
+
+Create a new repo file in this directory with the following contents.
```
Comment (GregSutcliffe) [inline extra/ovirt-node-plugin-foreman/README.md:136]: This doesn;t mention the filename that the next command assumes (`./foreman.repo`). Perhaps make this copyable bash like everything else? I.e:

```
cat > ./foreman.repo <<EOF
<repo contents>
EOF
```

File: extra/ovirt-node-plugin-foreman/README.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+        --debug \
+        ovirt-node-iso-*.iso
+    mv -f ovirt-node-iso-*.foreman.*.iso foreman.iso
+    test -d tftpboot && rm -rf tftpboot
+    livecd-iso-to-pxeboot foreman.iso
+
+Note we are renaming the target ISO to foreman.iso prior extracting kernel and
+initrd, otherwise you'd need to change kernel boot options.
+
+Finally, it's the dime to copy those files to TFTP server:
```
Comment (GregSutcliffe) [inline extra/ovirt-node-plugin-foreman/README.md]: Typo: it's the `time` to copy

File: README.ovirt_image.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+# Downloading existing Discovery PXE Image
+
+*Alpha version* of Foreman Discovery image (0.1.0) are [available for
+download] (http://lzap.fedorapeople.org/zzz/). These images were build in the
+production mode - root account is locked.
+
+To extract kernel and RAM disk do the following:
+
+    mv downloaded-image.iso foreman.iso
+    livecd-iso-to-pxeboot foreman.iso
```
Comment (GregSutcliffe) [inline README.ovirt_image.md]: We don't ship this tool, and we don't mention install ovirt tools on the tftp server - perhaps we should put the initrd0/vmlinuz on the download page?

File: README.tcl_image.md (hunk 4 above)
Comment (lzap) [inline README.tcl_image.md]: I think I was successful on RHEL6 so removing this requirement.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

File: extra/ovirt-node-plugin-foreman/README.md (hunk 6 above)
Comment (lzap) [inline extra/ovirt-node-plugin-foreman/README.md:136]: Oh yeah, sure.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

File: README.ovirt_image.md (hunk 8 above)
Comment (lzap) [inline README.ovirt_image.md]: Yeah, ISO makes no sense at all for discovery, I will change that to
distribute kernel/initrd separately.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

Comment (GregSutcliffe) [inline README.ovirt_image.md]: Leave the ISO up there - some people will want to use it :)

Comment (lzap) [inline README.ovirt_image.md]: Ok, all changes done, kernel/initrd are currently uploading from Brno
:-D

This should be ready to merge, then I am announcing that.

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman

File: README.ovirt_image.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+The process is described on a [separate wiki
+page](extra/ovirt-node-plugin-foreman/README.md).
+
+# Downloading existing Discovery PXE Image
+
+*Alpha version* of Foreman Discovery image (0.1.0) are [available for
+download] (http://lzap.fedorapeople.org/zzz/). These images were build in the
+production mode - root account is locked.
+
+To extract kernel and RAM disk do the following:
```
Comment (GregSutcliffe) [inline README.ovirt_image.md]: nitpick: you're not extracting anything now :P Maybe rephrase as "You can obtain the kernel & ramdisk from our mirror:"

File: README.ovirt_image.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+    cd /var/lib/tftpboot/boot
+    wget http://lzap.fedorapeople.org/zzz/discovery-initrd.img
+    wget http://lzap.fedorapeople.org/zzz/discovery-vmlinuz
+
+Check permissions and ownership after that.
+
```
Comment (GregSutcliffe) [inline README.ovirt_image.md:26]: Maybe add an optional line saying an ISO is also available for download if required?

**Comment by GregSutcliffe** [comment]
Merged as 88e3751, thanks @lzap!
